src/io.rs

```diff
@@ -993,7 +1003,12 @@ const FILE_ERROR: &wstr = L!("An error occurred while redirecting file '%ls'");
 const NOCLOB_ERROR: &wstr = L!("The file '%ls' already exists");
 
 /// Base open mode to pass to calls to open.
-const OPEN_MASK: libc::c_int = 0o666;
+const OPEN_MASK: Mode = Mode::S_IRUSR
+    .union(Mode::S_IWUSR)
+    .union(Mode::S_IRGRP)
+    .union(Mode::S_IWGRP)
+    .union(Mode::S_IROTH)
+    .union(Mode::S_IWOTH);
```

Comment:
This would be more readable as `Mode::from_bits(0o666).unwrap()` (but I don't know if that's stabilized or we would need to use a match and a panic in `const`).

Reply:
We can also do `from_bits_retain` that accepts unknown variants as well, therefore does not require an unwrap.

Reply:
Turns out we are on older version of nix that uses bitflags 1.0 rather than 2.0, so `from_bits_retain` is not available. I used `unasfe from_bits_unchecked` instead (does the same thing as from_bits_retain, only difference is that in 1.0 it used to be considered unsafe, and in 2.0 it no longer is), will remove it once we update.